File: src-tauri/src/commands/logs.rs
use tauri::State;

use crate::application::log_service;
use crate::application::log_service::{ExportLogsResponse, ListLogsResponse};
use crate::commands::{CleanupLogsPayload, ExportLogsPayload, ListLogsPayload};
use crate::AppState;

#[tauri::command]
pub async fn list_logs(
    state: State<'_, AppState>,
    payload: ListLogsPayload,
) -> Result<ListLogsResponse, String> {
    log_service::list(state.inner(), payload.limit, payload.offset)
        .await
        .map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn cleanup_logs(
    _state: State<'_, AppState>,
    payload: CleanupLogsPayload,
) -> Result<u64, String> {
    log_service::cleanup(payload.before_days)
        .await
        .map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn clear_all_logs(_state: State<'_, AppState>) -> Result<u64, String> {
    log_service::clear_all()
        .await
        .map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn export_logs_csv(
    _state: State<'_, AppState>,
    payload: ExportLogsPayload,
) -> Result<ExportLogsResponse, String> {
    log_service::export_csv(payload.provider_id, payload.limit, payload.offset, payload.file_path)
        .await
        .map_err(|e| e.to_string())
}

File: src/api/logs.ts
import { invoke } from "@tauri-apps/api/core";
import type { RequestLog } from "./types";

export interface LogsListResult {
  logs: RequestLog[];
  total: number;
  limit: number;
  offset: number;
}

export const logsApi = {
  list: (limit = 50, offset = 0): Promise<LogsListResult> =>
    invoke<LogsListResult>("list_logs", { payload: { limit, offset } }),

  cleanup: (beforeDays: number): Promise<number> =>
    invoke<number>("cleanup_logs", { payload: { beforeDays } }),

  clearAll: (): Promise<number> => invoke<number>("clear_all_logs"),

  exportCsv: (data: {
    provider_id?: string;
    limit?: number;
    file_path?: string;
  }): Promise<{ file_path: string; exported_count: number }> =>
    invoke<{ file_path: string; exported_count: number }>("export_logs_csv", {
      payload: data,
    }),
};
